cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsPlatformResources.java

```diff
@@ -901,6 +905,30 @@ public CloudEncryptionKeys encryptionKeys(ExtendedCloudCredential cloudCredentia
         return cloudEncryptionKeys;
     }
 
+    private Set<KeyListEntry> getListKeyResult(AmazonKmsClient client) {
+        String nextMarker = null;
+        Set<KeyListEntry> kmsKeys = new HashSet<>();
+        do {
+            ListKeysRequest listKeysRequest = new ListKeysRequest().withMarker(nextMarker);
+            ListKeysResult listKeysResult = client.listKeys(listKeysRequest);
+            kmsKeys.addAll(listKeysResult.getKeys());
+            nextMarker = listKeysResult.getNextMarker();
+        } while (nextMarker != null);
+        return kmsKeys;
+    }
+
+    private Set<AliasListEntry> getListAliasResult(AmazonKmsClient client) {
+        String aliasNextMarker = null;
+        Set<AliasListEntry> aliasListEntries = new HashSet<>();
+        do {
+            ListAliasesRequest listAliasesRequest = new ListAliasesRequest().withMarker(aliasNextMarker);
+            ListAliasesResult listAliasesResult = client.listAliases(listAliasesRequest);
+            aliasListEntries.addAll(listAliasesResult.getAliases());
```

Comment:
Like above, I wonder if `AliasListEntry` implements `hashCode()` and `equals()` properly, otherwise the usage of a `Set<>` for `aliasListEntries` does not really make sense.